docs/decisions/0003-instructor-dashboard-rest-api-architecture.rst

```diff
@@ -0,0 +1,276 @@
++--------+------------------------------------------------------------------------------------+--------------------------------------------+-----------+-------------------------------------------------------------------+
+| POST   | /api/instructor/v1/courses/{course_key}/cohorts                                    | openedx.core.djangoapps.course_groups      | Missing   | N/A                                                               |
++--------+------------------------------------------------------------------------------------+--------------------------------------------+-----------+-------------------------------------------------------------------+
+| PUT    | /api/instructor/v1/courses/{course_key}/cohorts/{cohort_id}                        | openedx.core.djangoapps.course_groups      | Missing   | N/A                                                               |
++--------+------------------------------------------------------------------------------------+--------------------------------------------+-----------+-------------------------------------------------------------------+
+| DELETE | /api/instructor/v1/courses/{course_key}/cohorts/{cohort_id}                        | openedx.core.djangoapps.course_groups      | Missing   | N/A                                                               |
++--------+------------------------------------------------------------------------------------+--------------------------------------------+-----------+-------------------------------------------------------------------+
+| POST   | /api/instructor/v1/courses/{course_key}/cohorts/{cohort_id}/students               | openedx.core.djangoapps.course_groups      | Missing   | N/A                                                               |
++--------+------------------------------------------------------------------------------------+--------------------------------------------+-----------+-------------------------------------------------------------------+
+| DELETE | /api/instructor/v1/courses/{course_key}/cohorts/{cohort_id}/students               | openedx.core.djangoapps.course_groups      | Missing   | N/A                                                               |
++--------+------------------------------------------------------------------------------------+--------------------------------------------+-----------+-------------------------------------------------------------------+
+| POST   | /api/instructor/v1/courses/{course_key}/cohorts/csv_upload                         | openedx.core.djangoapps.course_groups, util| Partial   | /courses/{course_id}/instructor/api/add_users_to_cohorts          |
++--------+------------------------------------------------------------------------------------+--------------------------------------------+-----------+-------------------------------------------------------------------+
+
+Discussion Management API
```

Comment:
Discussions also has an API that we can probably re-use and enhance with any gaps we need to fill: https://github.com/openedx/edx-platform/blob/master/docs/lms-openapi.yaml#L3752